dracut.sh

```diff
@@ -1062,7 +1062,9 @@ if ! [[ $outfile ]]; then
         mkdir -p "$dracutsysrootdir$efidir/Linux"
         outfile="$dracutsysrootdir$efidir/Linux/linux-$kernel${MACHINE_ID:+-${MACHINE_ID}}${BUILD_ID:+-${BUILD_ID}}.efi"
     else
-        if [[ -e "$dracutsysrootdir/boot/initrd-$kernel" ]]; then
+        if [[ -d "$dracutsysrootdir/boot/efi/${MACHINE_ID}" ]]; then
+            outfile="/boot/efi/${MACHINE_ID}/$kernel/initrd"
```

Comment:
I'd first try to submit this change upstream, because it's better not to diverge too much and this path is already referenced in `dracut-initramfs-restore.sh`.

https://github.com/dracutdevs/dracut/blob/79f9d9e1c29a9c8fc046ab20765e5bde2aaa3428/dracut-initramfs-restore.sh#L38

https://github.com/dracutdevs/dracut/blob/79f9d9e1c29a9c8fc046ab20765e5bde2aaa3428/dracut-initramfs-restore.sh#L46

If it isn't accepted, we can add it to SUSE only.

Reply:
I'm on FTO for a while. Could you handle it. Looking at the patch actually probably $MACHINE_ID needs to be checked first. it does so below.

Reply:
Ok, I will.